### DISCUSSION TITLE
按照文档中用大模型做为JudgeLLM进行客观评测的指导，运行时报错，SubjectiveSizePartitioner.__init__() got an unexpected keyword argument 'mode'，貌似是SubjectiveSizePartitioner这个类里面不支持mode这个初始化参数，请教下该怎么修改

### DISCUSSION THREAD

**Discussion by lilirm** [open-compass/opencompass#1632] [Q&A] [answered]
![WX20241022-172503@2x](https://github.com/user-attachments/assets/dfb09e67-202d-421a-bdf8-ad72b95dbc2c)

**Comment by bittersweet1999** [comment]
你参照的是哪个文档，可能out of date里之后需要更新一下，这个mode的参数被移到这里了（dataset的config里）：https://github.com/open-compass/opencompass/blob/be3c06a1587601ddb3658d3ad1079f735d8fd812/configs/datasets/subjective/alignbench/alignbench_v1_1_judgeby_critiquellm_new.py#L65

**Reply by lilirm** [reply]
我参考的是这个文档，可能是文档没及时更新，我试试你说的这个方式
![WX20241022-173115](https://github.com/user-attachments/assets/427432a3-9890-4cc5-96c7-54bc4b62494d)

**Reply by lilirm** [reply]
貌似不行，你发的这个文档应该是针对主观评测，我这边想对有答案的客观题（数学解答题）进行客观评测，使用大模型进行打分
![Uploading 1.png…]()

**Reply by bittersweet1999** [reply]
什么报错呢

**Reply by lilirm** [reply]
我的数据集类里面没有关于mode的使用，自己新建的一个简单读取数据的，所以会报错，和代码中给的主观评测代码样例有点不一样
![5](https://github.com/user-attachments/assets/fa59423e-9f16-4fe1-bbea-0e783453e088)

**Comment by bittersweet1999** [comment]
跟主观客观应该没有区别的呀，看下你的config呢

**Reply by lilirm** [reply]
我的数据集类就读取一下数据，，没有关于mode的配置
![2](https://github.com/user-attachments/assets/058a266d-dd4a-40f4-8038-caa4c262e10e)
![3](https://github.com/user-attachments/assets/19af5a84-0ff3-48b0-bd28-2c192ed1b364)

**Reply by bittersweet1999** [reply, marked as answer]
那你把`load(path:str, name:str)` 改成`load(path: str, name: str, *args, **kwargs):`应该就好了

**Reply by lilirm** [reply]
跑通了，感谢您的回答🙏
